pkg/resource/deploy/plan_executor.go

```diff
@@ -207,7 +224,17 @@ func (pe *planExecutor) performDeletes(ctx context.Context, opts Options) result
 
 	logging.V(7).Infof("performDeletes(...): beginning")
 
-	deleteSteps, res := pe.stepGen.GenerateDeletes(opts.DestroyTargets)
+	// At this point we have generated the set of resources above that we would normally want to
+	// delete.  However, if the user provided -target's we will only actually delete the specific
+	// resources that are in the set explicitly asked for.
+	var targetsOpt map[resource.URN]bool
+	if updateTargetsOpt != nil {
+		targetsOpt = updateTargetsOpt
+	} else if destroyTargetsOpt != nil {
+		targetsOpt = destroyTargetsOpt
+	}
```

Comment:
Any reason not to compute this in the caller to `performDeletes` and avoid needing to pass both values down to this layer?

Reply:
I actually started that way, but preferred that performDeletes not have any knowledge of what sort of operation we started with.  it just says "i want to compute the deletes as per allowed based on the targets specified".  

It also felt nice and clean to do all the targets-checking up front as well.  that way we can report immediately if someone specifies targets that don't exist *prior* to doing work.

Reply:
Why pass these in with separate parameters if we're just going to ignore one of them?